core/src/main/java/com/sequenceiq/cloudbreak/service/LoadBalancerConfigService.java

```diff
@@ -403,10 +405,14 @@ private NetworkV4Base getNetworkV4Base(Network network) {
     }
 
     private boolean isLoadBalancerEnabled(StackType type, String cloudPlatform, DetailedEnvironmentResponse environment, boolean flagEnabled) {
-        return getSupportedPlatforms().contains(cloudPlatform) &&
+        return getSupportedPlatforms().contains(cloudPlatform) && !isNoLoadBalancerInternal(environment) &&
             (flagEnabled || isLoadBalancerEnabledForDatalake(type, environment) || isLoadBalancerEnabledForDatahub(type, environment));
     }
 
+    private boolean isNoLoadBalancerInternal(DetailedEnvironmentResponse environment) {
+        return environment != null && environment.getTags() != null && environment.getTags().getTagValue(INTERNAL_DISABLE_TAG_KEY) != null;
```

Comment:
Please use environment.getTagValue(). This method already does the null check internally.